compiler/rustc_const_eval/src/const_eval/mod.rs

```diff
@@ -164,5 +165,31 @@ pub(crate) fn deref_mir_constant<'tcx>(
         },
     };
 
-    mir::ConstantKind::Val(op_to_const(&ecx, &mplace.into()), ty)
+    mir::ConstantKind::Val(op_to_const(&mut ecx, &mplace.into()), ty)
+}
+
+pub fn slice_for_alloc<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    alloc: ConstAllocation<'tcx>,
+) -> ConstAllocation<'tcx> {
+    let addr = tcx.create_memory_alloc(alloc);
```

Comment:
Please rename to `alloc_id` or so, the name currently does not reflect its type.